server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -0,0 +1,292 @@
+    public static final TransportVersion V_8_4_2 = new TransportVersion(8_04_02_99, "87c5b7b2-0f57-4172-8a81-b9f9a0198525");
+    public static final TransportVersion V_8_4_3 = new TransportVersion(8_04_03_99, "327cb1a0-9b5d-4be9-8033-285c2549f770");
+    public static final TransportVersion V_8_5_0 = new TransportVersion(8_05_00_99, "be3d7f23-7240-4904-9d7f-e25a0f766eca");
+    public static final TransportVersion V_8_5_1 = new TransportVersion(8_05_01_99, "d349d202-f01c-4dbb-85dd-947fb4267c99");
+    public static final TransportVersion V_8_5_2 = new TransportVersion(8_05_02_99, "b68b1331-fd64-44d9-9e71-f6796ec2024c");
+    public static final TransportVersion V_8_5_3 = new TransportVersion(8_05_03_99, "9ca3c835-e3b7-4622-a08e-d51e42403b06");
+    public static final TransportVersion V_8_5_4 = new TransportVersion(8_05_04_99, "97ee525c-555d-45ca-83dc-59cd592c8e86");
+    public static final TransportVersion V_8_6_0 = new TransportVersion(8_06_00_99, "e209c5ed-3488-4415-b561-33492ca3b789");
+    public static final TransportVersion V_8_6_1 = new TransportVersion(8_06_01_99, "9f113acb-1b21-4fda-bef9-2a3e669b5c7b");
+    public static final TransportVersion V_8_7_0 = new TransportVersion(8_07_00_99, "f1ee7a85-4fa6-43f5-8679-33e2b750448b");
+    /*
+     * READ THE JAVADOC ABOVE BEFORE ADDING NEW TRANSPORT VERSIONS
+     * Detached transport versions added below here. Starts at ES major version 10 equivalent.
+     */
+    // NOTE: DO NOT UNCOMMENT until all transport code uses TransportVersion
+    // public static final TransportVersion V_10_000_000 = new TransportVersion(10_000_000, "dc3cbf06-3ed5-4e1b-9978-ee1d04d235bc");
+    /*
+     * When adding a new transport version, ensure there is a gap (say, 100) between versions
+     * This is to make it possible to add intermediate versions for any bug fixes that may be required.
+     *
+     * When adding versions for patch fixes, add numbers in the middle of the gap. This is to ensure there is always some space
+     * for patch fixes between any two versions.
+     */
+
+    /** Reference to the current transport version */
+    public static final TransportVersion CURRENT = V_8_7_0;
+
+    /** Reference to the earliest compatible transport version to this version of the codebase */
+    // TODO: can we programmatically calculate or check this? Don't want to introduce circular ref between Version/TransportVersion
+    public static final TransportVersion MINIMUM_COMPATIBLE = V_7_17_0;
+
+    private static final Map<Integer, TransportVersion> idToVersion;
+
+    private static final TreeSet<TransportVersion> declaredVersions;
+
+    static {
+        Map<Integer, TransportVersion> builder = new HashMap<>();
+        Map<String, TransportVersion> uniqueIds = new HashMap<>();
+
+        Set<String> ignore = Set.of("ZERO", "CURRENT", "MINIMUM_COMPATIBLE");
+        for (Field declaredField : TransportVersion.class.getFields()) {
```

Comment:
Hardcoding TransportVerison makes this code difficult to unit test. Instead, this can be in a static method, with the Class passed in. Ultimately we don't care about the class, only that it has a bunch of public constants with a known naming structure of type TransportVersion.

Reply:
I don't follow. This static constructor is for populating the static maps, the assertion checks are a nice side-effect. We could move the assertion checks into a unit test?

Reply:
Ah, right. Added a test for the static init checks